### PR TITLE
Newsletters: add 74 (2019-11-27)

### PR DESCRIPTION
**Pull request by harding**
- [x] StackExchange section @bitschmidty 
- [x] Add Bitcoin Core release final URLs

Note, I may be unavailable to make updates after Tuesday 20:00 UTC, as I'll be traveling.  As always, feel free to push edits as necessary.

### CODE REVIEW

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+- [Bitcoin Core #17290][] changes which coin selection algorithm is used
+  in cases where the user requests certain inputs be used or asks that
+  the fee be selected from the payment amounts.  These now use Bitcoin
+  Core's normal default algorithm of Branch and Bound (BnB).  BnB was
+  designed to minimize fees and maximize privacy by optimizing for the
+  creation of changeless transactions.
+
+- [C-Lightning #3264][] includes several mitigations for [LND
+  #3728][], a bug in the implementation of gossip queries.  This
```
Comment (jonatack) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: Verified the repos and issue/PR numbers since this spans implementations.

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  change also adds two new command line parameters useful for testing
+  and debugging, `--hex` and `--features`.
```
Comment (jonatack) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: Too much detail for readers, but for info:
`--hex` (boolean arg to print out messages in hex)
`--features=<hex>` (to send specified features in init)

Found this [code comment](https://github.com/ElementsProject/lightning/pull/3264/files#diff-a9e88a70de2a2100cf545f044575e4efR724) interesting :eyes:

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    designed to help developers use the protocol safely.
+
+    Influenced by Jonas Nick's presentation on this topic at the Berlin
+    Lightning Conference, Adam Gibson wrote a separate [blog
+    post][gibson wagners] that describes Wagner's algorithm in much more
+    detail with a mix of math, intuitive analysis, and topical
+    information Bitcoiners may find interesting (such as the amusing
+    tidbit of [Wagner's paper][] citing both Adam Back and Wei Dai
+    several years before Nakamoto did [the same][bitcoin.pdf], though
+    for different work).  Anyone interested in the developing their own
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: the developing -> developing

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    cryptographic protocols is recommended to read both posts, as each
+    compliments the other without being repetitive about the subject.
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: compliments -> complements

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+## Selected Q&A from Bitcoin StackExchange
+
+*[Bitcoin StackExchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments of time to help curious or confused users.  In
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: spare moments of time -> spare moments

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+- [LND #3729][] makes it possible to generate invoices with millisatoshi
+  precision.  Previously, LND would not generate invoices with
+  sub-satoshi precision.
+
+- [LND #3499][] extends several RPCs, such as `listpayments` and
+  `trackpayments` to provide information about [multipath
+  payments][topic multipath payments], payments that can have multiple
+  parts that are sent over different routes.  These are not fully
+  supported by LND yet, but this merged PR makes it easier to add
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: to add -> to adding

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+- **New LND mailing list and new host of existing mailing lists:** a [new
+  mailing list][lnd engineering] hosted by Google Groups was announced
+  for LND application developers, with an [initial post][osuntokun lnd
+  plans] by Olaoluwa
+  Osuntokun describing short-term goals for LND's next release.
+  Separately, the existing mailing lists for [Bitcoin-Dev][] and
+  [Lightning-Dev][] have recently had their hosting [transferred][togami
+  ml update] to Oregon State University's [Open Source Lab][osl] (OSL), a
+  well-respected organization that provides hosting to a variety of open
+  source projects.  Optech extends our thanks to everyone involved in
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: It might be a good opportunity to explicitly call out Bryan and Warren who put in a lot of work on the mailing list.

**Comment by bitschmidty** [comment]
Added Stack Exchange section commit

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+  writing their own implementations of parts of the BIPs, including for
+  the libbitcoin and bcoin full verification nodes.
+
+    Two informative blog posts were also posted this week related to the
+    security of multiparty schnorr signatures.  Blockstream engineer
+    Jonas Nick [describes][nick musig] the [MuSig][] multiparty
+    signature scheme designed to allow users of [bip-schnorr][] to
+    aggregate multiple public keys into a single pubkey that they can
+    later sign for using a single signature generated collaboratively
+    among them.  Nick describes the three steps of the MuSig signing
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: I found this sentence a bit too long to parse easily ("that they can later sign for using..." was a bit tricky). Consider splitting it in two:

Blockstream engineer Jonas Nick describes the MuSig multiparty signature scheme. This scheme allows users of bip-schnorr to aggregate multiple public keys into a single pubkey and later sign for that aggregated pubkey using a single signature generated collaboratively among them.

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,301 @@
...
+    knows the transaction to sign is the nonce commitment.  (Not
+    mentioned in the blog post, but discussed on IRC, was that Pieter
+    Wuille and others have been investigating a construction based on a
+    Zero Knowledge Proof (ZKP) that could allow reduced interactivity.)
+    The blog post concludes with a suggestion that interested readers
+    review the MuSig implementation in [libsecp256k1-zkp][], which is
+    designed to help developers use the protocol safely.
+
+    Influenced by Jonas Nick's presentation on this topic at the Berlin
+    Lightning Conference, Adam Gibson wrote a separate [blog
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: +1 for including this. I always find waxwing's posts very clear and informative. :+1:

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+- [MuSig Signature Interactivity]({{bse}}91534)
+  Justinmoon asks why MuSig signing is always interactive and about secure,
+  offline interactive signing. Nickler explains each of the rounds involved
+  with MuSig signing as well as some pitfalls to avoid during signing.
+
+- [How does the bech32 length-extension mutation weakness work?]({{bse}}91602)
+  Jnewbery asks about a bech32 weakness where adding or removing q characters
+  immediately before the final p character of an address ending with p can make
+  a new, valid bech32 address, eluding the some of the bech32 error detection
+  properties. Pieter Wuille provides some algebraic details and MCCCS explains
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: > elusing some of the bech32 error detection properties

I'm not sure I'd phrase it this way. The error detection properties of bech32 work as designed _under the assumption that the length is fixed_. I'd either remove this entirely or reword it to something about "people's expectations of the error detection properties" or similar.

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+  However, if the LN user really wants to use the truncated chain, they
+  can start `lightningd` with the `--rescan` parameter to reprocess the
+  block chain.  <!-- yes, that parameter is for lightningd (not
+  bitcoind); see test in this PR -->
+
+- [Eclair #1221][] adds a `networkstats` API that returns various
+  information about the LN network as observed from the local node,
+  including the number of known channels, number of known LN nodes, the
+  capacity of LN nodes (grouped into percentiles), and the fees nodes are
+  changing (also grouped into percentiles).
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: changing or charging?

File: _posts/en/newsletters/2019-11-27-newsletter.md (hunk 6 above)
Comment (harding) [reply _posts/en/newsletters/2019-11-27-newsletter.md]: I don't think so?  (Your corrections are usually so good that I'm doubting my sanity here.)

Comment (jnewbery) [reply _posts/en/newsletters/2019-11-27-newsletter.md]: I don't think so either!

**Comment by harding** [comment]
@bitschmidty I pushed some edits to the B.SE section, including a significant edit in response to @jnewbery's comment here: https://github.com/bitcoinops/bitcoinops.github.io/pull/274#discussion_r350473452

Please feel free to drop the commit (in part or in whole) if you don't like it.

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 12]
```diff
@@ -109,7 +109,7 @@ Bitcoin infrastructure projects.
   [Lightning-Dev][] have recently had their hosting [transferred][togami
   ml update] to Oregon State University's [Open Source Lab][osl] (OSL), a
   well-respected organization that provides hosting to a variety of open
-  source projects.  Optech extends our thanks to everyone involved in
+  source projects.  Optech extends our thanks Warren Togami, Bryan Bishop, and everyone else involved in
```
Comment (jonatack) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: s/thanks/thanks to/

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 13]
```diff
@@ -243,10 +243,10 @@ endcomment %}
   information about the LN network as observed from the local node,
   including the number of known channels, number of known LN nodes, the
   capacity of LN nodes (grouped into percentiles), and the fees nodes are
```
Comment (jonatack) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: s/fees nodes/fees that nodes/ might be good for clarity/readability

File: _posts/en/newsletters/2019-11-27-newsletter.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+  offline interactive signing. Nickler explains each of the rounds involved
+  with MuSig signing as well as some pitfalls to avoid during signing.
+
+- [How does the bech32 length-extension mutation weakness work?]({{bse}}91602)
+  Jnewbery asks for details about why adding or removing q characters
+  immediately before the final p character of an address can sometimes
+  produce a new bech32 address that's valid even though [BIP173][] says
+  it "guarantees detection of **any error affecting at most 4
+  characters**" (emphasis in original).  This turns out to be a poor
+  choice of words in the BIP, as it was only meant to apply to
```
Comment (jnewbery) [inline _posts/en/newsletters/2019-11-27-newsletter.md]: > turns out to be a poor choice of words

This feels a bit strong (especially since this section is supposed to be reporting on what the question and answer are on SE, and the Q/A here is about the math, not design decisions). I also think we've covered the BIP 173 design decisions adequately in https://bitcoinops.org/en/newsletters/2019/11/13/#taproot-review-discussion-and-related-information.

I'd remove everything between "even though ... detect length-change errors."

File: _posts/en/newsletters/2019-11-27-newsletter.md (hunk 6 above)
Comment (adamjonas) [reply _posts/en/newsletters/2019-11-27-newsletter.md]: I read it over three times you're right. Direct quote from @jnewbery, "Jonas is open to adding grammatical errors to the newsletter."

**Comment by jnewbery** [comment]
ACK e9c0cbacb72dad3f2ca93c8e3802f429ebed352b.

Thanks as always @harding and @bitschmidty !

File: _posts/en/newsletters/2019-11-27-newsletter.md (hunk 10 above)
Comment (bitschmidty) [reply _posts/en/newsletters/2019-11-27-newsletter.md]: @jnewbery Thanks for that feedback. I think the changes made by @harding (thanks for those enhancements!) in 8f6841e9586692f1b85cd4e54b1f99a11355561a look good to me.

**Comment by bitschmidty** [comment]
ACK e9c0cbacb72dad3f2ca93c8e3802f429ebed352b 

Squashed

**Comment by bitschmidty** [comment]
Newsletter on time, ditty delay
Happy Thanksgiving, turkey day

Here comes Bitcoin Core 0.19
Had to make sure things were pristine

What is a bech32 length mutation?
`blockfilterindex` configuration

Bye Bye BIP37 and BIP70
But multiwallet mode now in the GUI

Progressing along in the taproot review group
Hop in ##taproot-bip-review on IRC to get the scoop

Thanks to @harding for another newsletter
and @jonatack @adamjonas @jnewbery for making it even better
